packages/server/src/insights/insights.controller.ts

```diff
@@ -72,35 +72,28 @@ export class InsightsController {
   @Get('list')
   @Roles(Role.PROFESSOR)
   async getAllInsights(): Promise<ListInsightsResponse> {
-    return Object.entries(INSIGHTS_MAP).map(([insightName, insight]) => ({
-      name: insightName,
-      displayName: insight.displayName,
-    }));
+    return this.insightsService.convertToInsightsListResponse(
+      Object.keys(INSIGHTS_MAP),
+    );
   }
 
   @Patch('')
   @Roles(Role.PROFESSOR)
   async toggleInsightOn(
     @Body() body: { insightName: string },
     @User() user: UserModel,
-  ): Promise<ListInsightsResponse> {
-    const updatedInsights = await this.insightsService.toggleInsightOn(
-      user,
-      body.insightName,
-    );
-    return updatedInsights;
+  ): Promise<void> {
+    await this.insightsService.toggleInsightOn(user, body.insightName);
+    return;
   }
 
   @Delete('')
   @Roles(Role.PROFESSOR)
   async toggleInsightOff(
     @Body() body: { insightName: string },
     @User() user: UserModel,
-  ): Promise<ListInsightsResponse> {
-    const updatedInsights = await this.insightsService.toggleInsightOff(
-      user,
-      body.insightName,
-    );
-    return updatedInsights;
+  ): Promise<void> {
+    await this.insightsService.toggleInsightOff(user, body.insightName);
```

Comment:
This is a JS question but since toggleInsightOff returns Promise<void> I think you can just say "return await this.insightsService..."

Reply:
You're right, that could have worked in the previous changes. 

Since the response was not being used anywhere I decided to have it return void instead.